app/scripts/helpers/GeometryWorker.js

```diff
@@ -0,0 +1,189 @@
+onmessage = function (m) {
+    process(m.data.geometryData);
+};
+
+process = function (data) {
+    var graphicData = data.graphicPrimitivesData;
+
+    graphicData.pointVertices = getFloatArray(graphicData.pointVertices);
+    graphicData.lineStripVertices = getFloatArray(graphicData.lineStripVertices);
+    graphicData.lineStripCounts = getIntArray(graphicData.lineStripCounts);
+    graphicData.triangleVertices = getFloatArray(graphicData.triangleVertices);
+    graphicData.triangleNormals = getFloatArray(graphicData.triangleNormals);
+
+    var geometries = [];
+    graphicData.numberOfCoordinates = 3;
+
+    addPoints(graphicData, geometries);
+    addTriangles(graphicData, geometries);
+    addCurves(graphicData, geometries);
+
+    sendChunked(geometries, data.nodeId)
+    
+    self.postMessage({msg: 'Complete', nodeId: data.nodeId});
```

Comment:
Send 'Complete' message to let main thread know where current geometry data ends.